File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/BinaryNumber/BinaryNumber_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.BinaryNumber.BinaryNumber import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        n = 5

        # when
        result = solution(n)

        # then
        self.assertEqual('101', result)

    def test_2(self):
        # given
        n = 1

        # when
        result = solution(n)

        # then
        self.assertEqual('1', result)

    def test_3(self):
        # given
        n = 29284

        # when
        result = solution(n)

        # then
        self.assertEqual('111001001100100', result)

    def test_4(self):
        # given
        n = 9438293

        # when
        result = solution(n)

        # then
        self.assertEqual('100100000000010001010101', result)

    def test_5(self):
        # given
        n = 326

        # when
        result = solution(n)

        # then
        self.assertEqual('101000110', result)

    def test_6(self):
        # given
        n = 84237182

        # when
        result = solution(n)

        # then
        self.assertEqual('101000001010101101101111110', result)


if __name__ == '__main__':
    unittest.main()

File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/BinaryNumber/BinaryNumber_test.js
import assert from 'assert'
import solution from './BinaryNumber.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const n = 5

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('101', result)
    done();
  });

  it('test 2', function (done) {
    // given
    const n = 1

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('1', result)
    done();
  });

  it('test 3', function (done) {
    // given
    const n = 29284

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('111001001100100', result)
    done();
  });

  it('test 4', function (done) {
    // given
    const n = 9438293

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('100100000000010001010101', result)
    done();
  });

  it('test 5', function (done) {
    // given
    const n = 326

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('101000110', result)
    done();
  });

  it('test 6', function (done) {
    // given
    const n = 84237182

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual('101000001010101101101111110', result)
    done();
  });
})
